drivers/sensor/Kconfig.sensor_clock

```diff
@@ -0,0 +1,30 @@
+# Sensor clock configuration options
+# Copyright(c) 2024 Cienet
+# SPDX-License-Identifier: Apache-2.0
+
+config SENSOR_CLOCK
+	bool
+	default y if SENSOR_ASYNC_API
```

Comment:
This will affect out-of-tree boards using the async API, so the new requirement to define `zephyr,sensor-clock` in the devicetree should be mentioned in the migration guide.

Reply:
Just as a note, with the current implementation if you leave out `zephyr,sensor-clock` you just get the system clock which is what was already used. So I think people don't have to do anything, but I 100% agree that this should be very well documented.